src/async/csharp/Program.cs

```diff
@@ -0,0 +1,141 @@
+
+
+// Connections are disposable because they hold system resources (ie a serial port or network connection).
+// In an asynchronous program we recommend the "await using" statement to automatically close and dispose
+// the Connection's resources when finished with it.
+// However, this is not available in .NET Framework programs, so in that case you can use the normal using
+// statement with a non-async connection open function.
+// If you cannot encapsulate all Zaber device use in a single code block like this, you should make sure
+// you manually close and dispose the Connection instance when your program is finished talking to devices.
+//
+// Note "await using" is only supported with ZML 5.0.0 or later under .NET or .NET Standard.
+// For other configurations remove both "await" statements from this line.
+await using (var connection = await Connection.OpenSerialPortAsync(PORT))
+{
+    // Enabling alerts speeds up detection of the end of device movement, but may cause problems
+    // for non-Zaber software communicating with the devices because it leaves them in a state
+    // where they can generate spontaneous messages. It is recommended if you are only using
+    // Zaber software (including the Zaber Motion Library).
+    connection.EnableAlerts();
+
+    // There is no async GetDevice because it just instantiates a device object
+    // without communicating with anything.
+    var xDevice = connection.GetDevice(X_DEVICE_ADDRESS);
+
+    // Everywhere you await a Zaber async function call, you have the option to do some other
+    // processing in parallel. Instead of using "await" immediately, you can assign the function's
+    // return value to a Task variable and then await it later after doing something else. You can
+    // also use Task.WhenAll() to wait for multiple async operations to complete.
+    var identifyTask = xDevice.IdentifyAsync();
+    // Could do something else here.
+    await identifyTask; // Block until IdentifyAsync() completes.
+    var xAxis = xDevice.GetAxis(X_AXIS_NUMBER);
+
+    Axis yAxis;
+    if (Y_DEVICE_ADDRESS == X_DEVICE_ADDRESS)
+    {
+        yAxis = xDevice.GetAxis(Y_AXIS_NUMBER);
+    }
+    else
+    {
+        var yDevice = connection.GetDevice(Y_DEVICE_ADDRESS);
+        await yDevice.IdentifyAsync();
+        yAxis = yDevice.GetAxis(Y_AXIS_NUMBER);
+    }
+
+    // Home the devices and wait until done.
+    // This is an example of overlapping async commands. Order of execution is not
+    // guaranteed and with more than two or three there is a risk of a device error.
+    await Task.WhenAll(xAxis.HomeAsync(), yAxis.HomeAsync());
+
+    // Grid scan loop
+    Axis[] axes = [xAxis, yAxis];
+    await foreach (var coords in Grid(X_GRID_POINTS, Y_GRID_POINTS))
+    {
+        // Here's one way of controlling the devices in parallel. This loop potentially sends the
+        // move commands and waits for the acknowledgements on different threads. There is a
```

Comment:
```suggestion
        // There is a
```
The sentence is irrelevant.